src/mixins/nr-affiliation-mixin.ts

```diff
@@ -91,32 +91,36 @@ export class NrAffiliationMixin extends Mixins(CommonMixin) {
   }
 
   /** Creates temporary business record and returns business identifier. */
-  private async createBusiness (accountId: number, nr: NameRequestI): Promise<string> {
+  private async createBusiness (
+    accountId: number, nr: NameRequestI, isNewAffiliation: boolean
+  ): Promise<string> {
     const createBusinessResponse =
-      await BusinessServices.createBusiness(this.getBusinessRequest(accountId, nr))
+      await BusinessServices.createBusiness(this.getBusinessRequest(accountId, nr)).catch(() => null)
 
-    if (createBusinessResponse.status === CREATED) {
-      return createBusinessResponse.data?.filing?.business?.identifier as string
-    }
+    if (createBusinessResponse?.status !== CREATED) {
+      // create failed -- delete new affiliation to avoid orphan records
+      if (isNewAffiliation) {
+        await AuthServices.removeNrAffiliation(accountId, nr.nrNum).catch(() => null)
+      }
 
-    // create failed, so delete the existing affiliation to avoid orphan records
-    await AuthServices.removeNrAffiliation(accountId, nr.nrNum).catch(() => null)
+      throw new Error('Unable to create new business')
+    }
 
-    throw new Error('Unable to create new business')
+    return createBusinessResponse.data?.filing?.business?.identifier as string
   }
 
   /** Returns business request object. */
   private getBusinessRequest (accountId: number, nr: NameRequestI): BusinessRequest {
     const name = this.isBenefitCompany(nr) ? 'incorporationApplication' : 'registration'
-    const legalType = this.isBenefitCompany(nr) ? 'BEN' : nr.entity_type_cd
+    const legalType = this.isBenefitCompany(nr) ? 'BEN' : nr.legalType
```

Comment:
For a DBA, entity_type_cd is "DBA" (which Filings UI can't use), but legalType is "SP" (which works).